authors/apps/authentication/views.py

```diff
@@ -73,3 +93,153 @@ def update(self, request, *args, **kwargs):
 
         return Response(serializer.data, status=status.HTTP_200_OK)
 
+
+class PasswordResetView(GenericAPIView):
+    """
+    Sends password reset email with id token
+    """
+    permission_classes = (AllowAny,)
+
+    def post(self, request):
+        response = None
+        response_status = None
+        errors = {}
+        data = request.data.get("user")
+        email = data.get("email")
+        if not email:
+            errors["email"] = ["this field is required"]
+            response_status = status.HTTP_406_NOT_ACCEPTABLE
+        else:
+            mail_validator = EmailValidator()
+            try:
+                mail_validator(email)
+            except ValidationError as error:
+                errors["email"] = error
+                response_status = status.HTTP_400_BAD_REQUEST
+            else:
+                try:
+                    user = User.objects.get(email=email)
+                except ObjectDoesNotExist:
+                    errors["email"] = ["no account with that email address"]
+                    response_status = status.HTTP_406_NOT_ACCEPTABLE
+                else:
+                    token, created = Token.objects.get_or_create(user=user)
+                    if not created:
+                        token.created = timezone.now()
```

Comment:
@Ogutu-Brian you could implement the sending of email functions in one file, say mailer.py file, from where all email sending functions would be called. You and @Bakley could work on this file together. This keeps our views cleaner and implements good coding practices.

Reply:
Let me do that @IssaIan

Reply:
It is done